echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/pagerduty/PagerDutyNotificationService.groovy

```diff
@@ -50,17 +58,91 @@ class PagerDutyNotificationService implements NotificationService {
 
   @Override
   void handle(Notification notification) {
+    Map details = new LinkedHashMap()
+    details.put("details", notification.additionalContext.details)
+
+    Map<String, String> errors = new HashMap()
+    Map<String, String> pdErrors = new HashMap()
+
     notification.to.each {
-      pagerDuty.createEvent(
-        "Token token=${token}",
-        new PagerDutyService.PagerDutyCreateEvent(
-          service_key: it,
-          client: "Spinnaker (${notification.source.user})",
-          description: notification.additionalContext.message
+      try {
+        Map response = pagerDuty.createEvent(
```

Comment:
Value in typing the response from `pagerDuty.createEvent()` to something better than `Map`?